distributed/tests/test_worker.py

```diff
@@ -2009,6 +2015,104 @@ async def test_process_executor(c, s, a, b):
         assert (await future) != os.getpid()
 
 
+@gen_cluster(client=True, nthreads=[])
+async def test_annotator_choose_executor(c, s):
+    """Test that it is possible for a task to choose executor"""
+
+    def get_thread_name(x):
+        return threading.current_thread().name
+
+    def set_executor_by_task_output(ts: TaskState, value: object) -> dict:
+        return {"annotations": {"executor": value}, "dependents": True}
+
+    def f(x):
+        return x
+
+    async with Worker(
+        s.address,
+        nthreads=1,
+        executor={
+            "exec1": ThreadPoolExecutor(1, thread_name_prefix="Executor1"),
+            "exec2": ThreadPoolExecutor(1, thread_name_prefix="Executor2"),
+        },
+        annotators=[set_executor_by_task_output],
+    ):
+        dsk = {
+            "f1": (f, "exec1"),
+            "g1": (get_thread_name, "f1"),
+            "f2": (f, "exec2"),
+            "g2": (get_thread_name, "f2"),
+        }
+        res1, res2 = c.get(dsk, ["g1", "g2"], sync=False, asynchronous=True)
+        assert "Executor1" in await res1
+        assert "Executor2" in await res2
```

Comment:
The fact that you had to use get/compute here is intersting.  There are interesting challenges with doing this with futures / dependencies that might change.  I still think that this is a useful feature to have, but it seems like this approach might not be comprehensive if we want to solve things across update_graph calls.  Agree or disagree?

Reply:
Agree! I hadn't thought about this issue before today. As you say, it is still useful but I think we should put this PR on hold for a bit and see if we can come up with a better approach.